utils/telemetry/service.go

```diff
@@ -0,0 +1,46 @@
+package telemetry
+
+import (
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// REALMC-7243 same as the new cli Segment tracking
+
+type Service struct {
+	userID      string
+	executionID string
+	command     string
+	tracker     Tracker
+}
+
+func (s *Service) SetFields(telemetryOn bool, userID string, command string) {
```

Comment:
so I know I previously mentioned having `SetCommand` and `SetUserID` methods, but I can see `SetFields` was doing even more than just that now.

I still think `SetUserID` should be its own call, so we can do that separately, down the line in the `run` implementation (and still have the `tracker` and `command` set).  Instead of `SetCommand`, let's have a `Setup(command string)` function, and it can also do the `tracker` logic you have here (based on the `s.NoTelemetry` field)